### PR TITLE
add cylindrical cutouts to *_bins.scad

### PR DESCRIPTION
**Pull request by jannessm**
I added cylindrical cutouts which are placed in a linear pattern to easily support more advanced bins.

### CODE REVIEW

**Comment by Ruudjhuu** [comment]
This is a nice idea.

Could you put most of the logic in a new module in the utility file? Then the function you adapted is still very simple. 
Add two booleans to enable the "normal" cutouts and the cylindrical cutouts and make some logic to pick the correct cutout module. Also add some error handling, (if both booleans are enabled report a message to the user that this is probably not what they want).

File: gridfinity-rebuilt-bins.scad [hunk 1]
```diff
@@ -73,11 +87,35 @@ div_base_y = 0;
 color("tomato") {
 gridfinityInit(gridx, gridy, height(gridz, gridz_define, style_lip, enable_zsnap), height_internal) {
 
-    if (divx > 0 && divy > 0)
-    cutEqual(n_divx = divx, n_divy = divy, style_tab = style_tab, scoop_weight = scoop);
+    if (divx > 0 && divy > 0) {
+        
+        cutEqual(n_divx = divx, n_divy = divy, style_tab = style_tab, scoop_weight = scoop);
+    
+    } else if (cdivx > 0 && cdivy > 0) {
```
Comment (Ruudjhuu) [inline gridfinity-rebuilt-bins.scad:94]: Move logic to module in utility and use that module here so the high level function is kept simple.

**Review by Ruudjhuu — CHANGES_REQUESTED** [review]

File: gridfinity-rebuilt-bins.scad (hunk 1 above)
Comment (jannessm) [reply gridfinity-rebuilt-bins.scad:94]: Done!

**Review by Ruudjhuu — APPROVED** [review]

**Comment by Ruudjhuu** [comment]
For the next time, please re-request a review. I missed the latest change and now other pull requests are merged which results in conflicts.

**Pull request merged**